### DISCUSSION TITLE
Support multiple topic pattern in KafkaListener

### DISCUSSION THREAD

**Discussion by rajjaiswalsaumya** [spring-projects/spring-kafka#2782] [Q&A] [answered]
Do we support multiple `topicpattern` in KafkaListener ? I see `topicPattern` is `String` while the `topics` is `String array`. Unfortunately, in our case, topics are created by customers so microservice does not know the topic name beforehand. So we use a pattern for customer topic creation in kafka but the same listener does not listen to the topics as `topics` does not support a pattern and requires an exact topic name while `topicpattern` accepts a single topic pattern.

**Comment by garyrussell** [comment, marked as answer]
Yes, but the pattern is a Regular Expression (RegEx) so you can use `(foo.*|bar.*)` or similar to match multiple patterns.

**Reply by garyrussell** [reply]
Also, this is a limitation of the underlying `kafka-clients` API, which only takes a single pattern.

**Reply by rajjaiswalsaumya** [reply]
Oh okay, thanks Gary.

**Reply by rajjaiswalsaumya** [reply]
This worked, Closing discussion.
